components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -225,16 +219,6 @@ private RegistrationResponseProfile createOAuthApplication(RegistrationRequestPr
                 log.debug("Created OAuth App " + applicationName);
             }
 
-            try {
-                if (isHashDisabled) {
-                    createdApp = oAuthAdminService
-                            .getOAuthApplicationDataByAppName(oAuthConsumerApp.getApplicationName());
-                }
-            } catch (IdentityOAuthAdminException e) {
-                throw IdentityException.error(DCRException.class, ErrorCodes.BAD_REQUEST.toString(), e.getMessage());
-
-            }
-
             if (log.isDebugEnabled()) {
                 log.debug("Retrieved Details for OAuth App " + createdApp.getApplicationName());
```

Comment:
There is a possible NPE in this line. What if we get the identity exception before createdApp is initialized?

Reply:
If we get the identity exception, the exception will be thrown. so it will not reach this line.